File: lib/schemas/documentation.ts
import { z } from "zod";

export const DocConnector = z.object({
  kind: z.string(),
  label: z.string(),
  available: z.boolean(),
});
export type DocConnector = z.infer<typeof DocConnector>;

export const PrepareDocResult = z.object({
  status: z.string(),
  mode: z.enum(["branch", "pr"]),
  repo_name: z.string(),
  ado_project: z.string(),
  branch: z.string(),
  pr_id: z.string().nullable().optional(),
  pr_title: z.string().nullable().optional(),
  head_sha: z.string(),
  languages: z.array(z.string()).default([]),
  upstream_summary: z.string().default(""),
});
export type PrepareDocResult = z.infer<typeof PrepareDocResult>;

export const DocType = z.enum([
  "doc_set", "overview", "sdd", "api_reference", "code_summary",
  "changelog", "release_notes", "rtm", "run_summary", "compliance", "custom",
]);
export type DocType = z.infer<typeof DocType>;

export const GeneratedDoc = z.object({
  id: z.string(),
  type: z.string().default("custom"),
  title: z.string().default(""),
  filename: z.string().default(""),
  format: z.string().default("md"),
  path: z.string().default(""),
  contents: z.string().default(""),
  bytes: z.number().default(0),
});
export type GeneratedDoc = z.infer<typeof GeneratedDoc>;

export const DocSetResponse = z.object({
  documents: z.array(GeneratedDoc).default([]),
  doc_list: z
    .array(z.object({
      id: z.string(), type: z.string().default("custom"), title: z.string().default(""),
      filename: z.string().default(""), format: z.string().default("md"), bytes: z.number().default(0),
    }))
    .default([]),
  pr_url: z.string().nullable().optional(),
  context: z.object({
    repo_name: z.string().default(""),
    ado_project: z.string().default(""),
    mode: z.enum(["branch", "pr"]).default("branch"),
    source_branch: z.string().default(""),
    pr_id: z.string().nullable().optional(),
    head_sha: z.string().default(""),
    languages: z.array(z.string()).default([]),
    upstream_summary: z.string().default(""),
  }),
});
export type DocSetResponse = z.infer<typeof DocSetResponse>;

File: backend/shared/models/documentation.py
"""Standalone Documentation agent artifact (session-held doc set).

Distinct from the pipeline-mode `DocumentationArtifact` planned for
shared/models/artifacts.py — this is the interactive standalone surface: the agent
generates documents on demand, each saved to the local docs folder (Azure Blob
later) and surfaced in the page's left-side list.
"""
from __future__ import annotations

from typing import List, Literal, Optional

from pydantic import BaseModel, Field

DocType = Literal[
    "doc_set", "overview", "sdd", "api_reference", "code_summary",
    "changelog", "release_notes", "rtm", "run_summary", "compliance", "custom",
    "runbook_update", "knowledge_article",
]


class GeneratedDoc(BaseModel):
    id: str = ""
    type: DocType = "custom"
    title: str = ""
    filename: str = ""
    format: str = "md"
    path: str = ""              # local path on disk (or blob URL later)
    contents: str = ""          # markdown body (served to the viewer)
    bytes: int = 0
    diff: Optional[str] = None          # runbook_update: unified diff of the changed sections
    source_ref: Optional[str] = None    # where the prior version was read from (wiki page path / SharePoint item)
    issue_ref: Optional[str] = None     # knowledge_article: the fixed issue this article is about


class DocContext(BaseModel):
    repo_name: str = ""
    ado_project: str = ""
    mode: Literal["branch", "pr"] = "branch"
    source_branch: str = ""
    pr_id: Optional[str] = None
    head_sha: str = ""
    languages: List[str] = Field(default_factory=list)
    upstream_summary: str = ""


class DocumentationArtifact(BaseModel):
    context: DocContext = Field(default_factory=DocContext)
    documents: List[GeneratedDoc] = Field(default_factory=list)
    pr_url: Optional[str] = None
    status: Literal["idle", "generating", "ready", "pr_opened"] = "idle"
